### DISCUSSION TITLE
Make participants click and hold mouse for a certain amount of time

### DISCUSSION THREAD

**Discussion by aislinnkeogh** [jspsych/jsPsych#3009] [Q&A]
I'm trying to implement an experiment where it takes participants longer to give longer words as responses. The way I want to do this is by having them click a button and hold their mouse down for 500ms per letter in the word on that button. If they release the button too early, they have to start again. And there needs to be some kind of visual feedback as to how long they still need to hold the mouse e.g. the word appearing letter-by-letter above the button they're clicking.

Does this seem like the kind of thing I could achieve with the mouse-tracking extension? My vague idea was to use a loop function that keeps running until the minimum clicking time has been reached, and checks at every step (letter in the word) to make sure the mouse is still down. But it seems like the mouse-tracking extension can only handle mouseup/mousedown *events*, rather than enabling you to query the state of the mouse at any given moment, so I'm not sure this will work. Open to any other ideas!

**Comment by jodeleeuw** [comment]
Hey @aislinnkeogh. 

This is probably best done with a custom plugin. You can then listen to the mouse events directly and implement the visual feedback. I think it might take more work to hack a solution together with existing plugins than to just make a new one. 

[There's a guide here on plugin development](https://www.jspsych.org/7.3/developers/plugin-development/). Feel free to follow up with troubleshooting questions!
